plugins/chat/spec/jobs/regular/mark_all_channel_threads_read_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+RSpec.describe Jobs::Chat::MarkAllChannelThreadsRead do
+  fab!(:channel) { Fabricate(:chat_channel, threading_enabled: true) }
+
+  context "when enable_experimental_chat_threaded_discussions is false" do
+    before { SiteSetting.enable_experimental_chat_threaded_discussions = false }
+
+    it "does nothing" do
+      Chat::Channel.any_instance.expects(:mark_all_threads_as_read).never
+      described_class.new.execute(channel_id: channel.id)
+    end
+  end
+
+  context "when enable_experimental_chat_threaded_discussions is true" do
+    fab!(:thread_1) { Fabricate(:chat_thread, channel: channel) }
+    fab!(:thread_2) { Fabricate(:chat_thread, channel: channel) }
+    fab!(:user_1) { Fabricate(:user) }
+    fab!(:user_2) { Fabricate(:user) }
+    fab!(:thread_1_message_1) { Fabricate(:chat_message, thread: thread_1) }
+    fab!(:thread_1_message_2) { Fabricate(:chat_message, thread: thread_1) }
+    fab!(:thread_1_message_3) { Fabricate(:chat_message, thread: thread_1) }
+    fab!(:thread_2_message_1) { Fabricate(:chat_message, thread: thread_2) }
+    fab!(:thread_2_message_2) { Fabricate(:chat_message, thread: thread_2) }
+
+    before do
+      SiteSetting.enable_experimental_chat_threaded_discussions = true
+      channel.add(user_1)
+      channel.add(user_2)
+      thread_1.add(user_1)
+      thread_2.add(user_2)
+    end
+
+    it "marks all threads as read across all users in the channel" do
```

Comment:
It would be better to have a test checking we are changing from unread=1 to unread=0,  otherwise if for some reason it's already 0 before due to another bug, we wouldnt catch regression here.

Reply:
Good point, will add.